Small change to design token naming scheme (#1947)
ni/nimble: packages/nimble-components/CONTRIBUTING.md

Public names for theme-aware tokens are specified in `src/theme-provider/design-token-names.ts`. Use the following structure when creating new tokens.

> `[element]-[part]-[interaction_state]-[other_state]-[token_type]`

Reviewer: ```suggestion
`[element]-[part]-[interaction_states]-[remaining_states]-[token_type]`
```

1. Where **element** is the type to which the token applies (e.g. 'application', 'body', or 'title-plus-1').
2. Where **part** is the specific part of the element to which the token applies (e.g. 'border', 'background', or shadow).

> 3. Where **state** is one or more of the following interaction states: 'active', 'disabled', 'hover', or 'selected'. Multiple states should be sorted alphabetically.

Reviewer 2: nit: I prefer the "(e.g. ...)" format we had before since this may not be an exhaustive list. "Focused" and its variants are the only ones I can think of that might be missing, but I prefer to keep some wiggle room if a new state arises or we're forgetting one.

> 4. Where **variant** is any other distinguishing descriptor (e.g. 'accent', 'primary, or 'large'). Multiple variants should be sorted alphabetically.

Reviewer: "state" vs "variant" / "distinguishing descriptor" does not have a clear concept to me. i.e. it's not clear what it is a variant of. The examples are more like variants of the element/part not variants of the interaction state. If it's modelled after the appearance variant concept it's clearer that in that context it's a variant of appearance.

I'd propose `interaction state` vs `appearance state`. I think the examples show how it's not over constrained to the appearance attribute concepts (i.e. by including 'large')

As an aside this kind of also parallels the [cascade guidance](https://github.com/ni/nimble/blob/main/packages/nimble-components/docs/css-guidelines.md#prefer-cascade-for-state-changes) where the interaction states for the most part are before other states.

Author: My intent was for "variant" to be a catch-all for however else we're distingushing this token from another (with the same element, part, and/or interaction state). It may not be an appearance-based distinction. As a contrived example, suppose our token is a distinct width or font property that we want to use when the locale is German (because German words tend to be longer and take up more space). That doesn't seem like an "appearance state" to me.

As a counter proposal, I'm changing these to `interaction_state` and `other_state`, but I'm happy to use something else if you don't like `other_state`.

Reviewer: gave an alternative and approved assuming we are aligned, if not can give a counter

> 3. Where **interaction_state** is one or more interaction states (e.g. 'active', 'disabled', 'hover', or 'selected'). Multiple values should be sorted alphabetically.
> 4. Where **other_state** is any other distinguishing state (e.g. 'accent', 'primary, or 'large'). Multiple values should be sorted alphabetically.

Reviewer: ```suggestion
3. Where **interaction_states** is one or more interaction states (e.g. 'active', 'disabled', 'hover', or 'selected'). Multiple values should be sorted alphabetically.
4. Where **remaining_states** the remaining, non-interaction states (e.g. 'accent', 'primary, or 'large'). Multiple values should be sorted alphabetically.
```
